### PR TITLE
Added clientside hologram parenting

### PR DESCRIPTION
**Pull request by adamnejm**
Tested briefly, garbage collection works, changing parents, unparenting, hierarchy recursion works.
Code check and more testing probably required.

Added CL privilege hologram.setParent.
Engine flag on the parent - [EFL_FORCE_CHECK_TRANSMIT](https://wiki.garrysmod.com/page/Enums/EFL) might be required for some entities, but I'm not adding that since I can't test that on anything.

### CODE REVIEW

File: lua/starfall/libs_sh/holograms.lua [hunk 1]
```diff
@@ -434,6 +434,78 @@ else
...
+			holo.sf_parent = parent
+			
+			holo:SetParent(parent, attachment)
+			
+		else
+			
+			if holo.sf_parent then
+				holo.sf_parent.sf_children[holo] = nil
+			end
+
```
Comment (Damianu) [inline lua/starfall/libs_sh/holograms.lua]: holo.sf_parent = nil

**Comment by thegrb93** [comment]
I don't think the permission needs to check prop ownership.

**Comment by thegrb93** [comment]
I'm testing it now

**Comment by thegrb93** [comment]
Oh I'm dumb, that keeps you from parenting other's holograms. I'll revert that.

**Pull request merged**